lib/dynamo.js

```diff
@@ -10,29 +11,32 @@ const clientOptions = { region, httpOptions, maxRetries }
 // NOTE: must use STS in the region containing the lambda, not the DDB table
 // TODO: shouldn't this happen automagically, with clientOptions.region set?
 const myRegion = process.env.AWS_REGION || 'us-east-1'
-const endpoint = new AWS.Endpoint(`sts.${myRegion}.amazonaws.com`)
-const sts = new AWS.STS({ endpoint, httpOptions, maxRetries })
+const sts = new STSClient({ region: myRegion, httpOptions, maxRetries })
 
 // optionally load a client using a role
 exports.client = async () => {
   if (process.env.ARRANGEMENTS_DDB_ACCESS_ROLE) {
     const RoleArn = process.env.ARRANGEMENTS_DDB_ACCESS_ROLE.split(',')[0]
     const RoleSessionName = 'dovetail-counts-lambda-dynamodb'
+    const command = new AssumeRoleCommand({ RoleArn, RoleSessionName });
+
     try {
-      const data = await sts.assumeRole({ RoleArn, RoleSessionName }).promise()
+      const data = await sts.send(command)
       const { AccessKeyId, SecretAccessKey, SessionToken } = data.Credentials
-      return new AWS.DynamoDB({
+      return new DynamoDBClient({
         ...clientOptions,
```

Comment:
Will also need to change `clientOptions` to use the `requestHandler` instead of `httpOptions`.